src/Our.Umbraco.Ditto/ComponentModel/ValueResolvers/AppSettingValueResolver.cs

```diff
@@ -28,12 +28,21 @@ public class AppSettingValueResolver : DittoValueResolver<AppSettingAttribute>
         /// </returns>
         public override object ResolveValue(ITypeDescriptorContext context, AppSettingAttribute attribute, CultureInfo culture)
         {
-            if (string.IsNullOrWhiteSpace(attribute.AppSettingKey))
+            // default to attribute value
+            string appSettingKey = attribute.AppSettingKey;
```

Comment:
I can't think of any reason PropertyName would ever be null, so you could probably just use

`var key = attribute.PropertyName ?? context.PropertyDescriptor;`